dev/scripts/build-docker.sh

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+#
+# The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+# (the "License"). You may not use this work except in compliance with the License, which is
+# available at www.apache.org/licenses/LICENSE-2.0
+#
+# This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied, as more fully set forth in the License.
+#
+# See the NOTICE file distributed with this work for information regarding copyright ownership.
+#
+
+#
+# This script generates a Docker image for the repository from its current state.
+#
+
+set -e
+
+readonly SCRIPT_DIR=$(cd "$( dirname "$0" )"; pwd)
+readonly DOCKER_DIR="${SCRIPT_DIR}/../../integration/docker"
+readonly GENERATE_TARBALL_SCRIPT="${SCRIPT_DIR}/generate-tarball.sh"
```

Comment:
Should we use `readonly` in our other shell scripts as well? If so, could you open a ticket?